ToDoList/ViewModel/WeatherViewModel.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  WeatherViewModel.swift
+//  ToDoList
+//
+//  Created by Nathi Mabena on 2025/08/21.
+//
+
+import Foundation
+
+@MainActor
+final class WeatherViewModel: ObservableObject {
+    @Published var currentWeather: Weather?
+    @Published var forecast: [Forecast] = []
+    @Published var isLoading = false
+    @Published var errorMessage: String?
```

Comment:
```suggestion
    @Published var errorMessage: String? = nil
```

default error message to nil here so you don't have to set it to nil in the function